src/openloco/audio/audio.cpp

```diff
@@ -777,11 +778,75 @@ namespace openloco::audio
         return false;
     }
 
-    static void sub_48A274(vehicle* v)
+    static void sub_48A274(vehicle* baseVehicle)
     {
-        registers regs;
-        regs.esi = (int32_t)v;
-        call(0x0048A274, regs);
+        // TODO: move vehicle_26 cast up
+        auto v = baseVehicle->as_vehicle_2or6();
+        if (v == nullptr)
+            return;
+
+        if (v->sound_id == 0xFF)
+            return;
+
+        // TODO: left or top?
+        if (v->sprite_left == (int16_t)0x8000u)
+            return;
+
+        if (_112C666 >= config::get().max_vehicle_sounds)
+            return;
+
+        auto spritePosition = viewport_pos(v->sprite_left, v->sprite_top);
+
+        auto main = WindowManager::getMainWindow();
+        if (main != nullptr && main->viewports[0] != nullptr)
+        {
+            auto viewport = main->viewports[0];
+            ViewportRect extendedViewport = {};
+
+            auto quarterWidth = viewport->view_width / 4;
+            auto quarterHeight = viewport->view_height / 4;
+            extendedViewport.left = viewport->view_x - quarterWidth;
+            extendedViewport.top = viewport->view_y - quarterHeight;
+            extendedViewport.right = viewport->view_x + viewport->view_width + quarterWidth;
+            extendedViewport.right = viewport->view_y + viewport->view_height + quarterHeight;
+
+            if (extendedViewport.contains(spritePosition))
+            {
+                // jump + return
+                _112C666 += 1;
+                v->var_4A |= 1;
+                v->var_4E = (uint8_t)main->type;
```

Comment:
Can you look to see how 4e is used maybe can change the type

Reply:
I expect in the vehicle update code it uses this to work out the viewport location to play the sound. I'll have to compare it to my code.